### PR TITLE
cleaning up http spec

### PR DESCRIPTION
**Pull request by jiyoontbd**
## changes
* removed Exceptions as we're not conforming to exception types in tbdex error message struct
* removed `Authentication` and `Authorization` headers for each protected endpoint description and added reference to top level `Authentication` heading that explains how bearer did authn works
* Cleaning up tables
* Removed wording in discoverability section about how serviceEndpoint should resolve to either API or DID, which implied that one must recursively resolve til they get a publicly addressable API URL. It is confusing and probably not something required by DID spec
* Added example around idempotency key
* Added explanation around `serviceEndpoint`

### CODE REVIEW

**Review by mistermoe — APPROVED** [review]

**Review by KendallWeihe — APPROVED** [review]

File: specs/http-api/README.md [hunk 1]
```diff
@@ -73,12 +70,13 @@ Version: Draft
...
-| ----------------- | ------------------------------------------------------------------------------------------- |
-| `id`              | see [DID-CORE spec](https://www.w3.org/TR/did-core/#services)                               |
-| `type`            | `PFI`                                                                                       |
-| `serviceEndpoint` | PFI's publicly addressable API endpoint or DID which has PFIs publicly addressable endpoint |
-If the serviceEndpoint is itself a DID, this DID should resolve to a document and then its serviceEndpoints can be examined for the `PFI` type entry. The ID can be chosen at the discretion of the PFI, but the service entry should be of type `PFI`.
+| Property          | Value                                                         |
+| ----------------- | ------------------------------------------------------------- |
+| `id`              | see [DID-CORE spec](https://www.w3.org/TR/did-core/#services) |
+| `type`            | `PFI`                                                         |
+| `serviceEndpoint` | PFI's publicly addressable API endpoint                       |
```
Comment (decentralgabe) [inline specs/http-api/README.md:77]: could we link this API endpoint to something?

File: specs/http-api/README.md [hunk 2]
```diff
@@ -166,14 +147,37 @@ Pagination is supported using the following query params:
...
+  "metadata": {
+    "from": "did:key:z6MkvUm6mZxpDsqvyPnpkeWS4fmXD2jJA3TDKq4fDkmR5BD7",
+    "to": "did:ex:pfi",
+    "exchangeId": "rfq_01ha83pkgnfxfv41gpa44ckkpz",
+    "kind": "order",
+    "id": "order_01ha83pkgsfk6t1kxg7q42s48j",
+    "createdAt": "2023-09-13T20:28:40.345Z",
+    "protocol": "1.0"
+  },
+  "data": {},
```
Comment (decentralgabe) [inline specs/http-api/README.md:163]: is data present even if it's a null value?

File: specs/http-api/README.md [hunk 3]
```diff
@@ -166,14 +147,37 @@ Pagination is supported using the following query params:
...
+```json
+{
+  "metadata": {
+    "from": "did:key:z6MkvUm6mZxpDsqvyPnpkeWS4fmXD2jJA3TDKq4fDkmR5BD7",
+    "to": "did:ex:pfi",
+    "exchangeId": "rfq_01ha83pkgnfxfv41gpa44ckkpz",
+    "kind": "order",
+    "id": "order_01ha83pkgsfk6t1kxg7q42s48j",
+    "createdAt": "2023-09-13T20:28:40.345Z",
+    "protocol": "1.0"
```
Comment (decentralgabe) [inline specs/http-api/README.md:161]: should this now be 2.0?

File: specs/http-api/README.md (hunk 2 above)
Comment (jiyoontbd) [reply specs/http-api/README.md:163]: yep! order's data is empty.

File: specs/http-api/README.md (hunk 1 above)
Comment (jiyoontbd) [reply specs/http-api/README.md:77]: 🤔 like to a spec?

File: specs/http-api/README.md [hunk 4]
```diff
@@ -127,30 +125,13 @@ If the serviceEndpoint is itself a DID, this DID should resolve to a document an
...
-
-
 # Query Params
 Query parameters, also known as query strings, are a way to send additional information to the server as part of a URL. They allow clients to provide specific input or customize the server's response. Query parameters typically follow the main URL and start with a `?` character. They consist of key-value pairs, and multiple pairs can be separated by `&` characters
 
 Query params are supported by many of the `GET /${resource}` endpoints in the following ways
 
-* Simple Example: `?simple=field`
-* Same Field; Multiple Values: `field=value&field=anotherValue`
+* Simple Example: `?field=foo`
```
Comment (angiejones) [inline specs/http-api/README.md:133]:
```suggestion
* Simple Example: `?field=simpleValue`
```

foo/bar is frowned upon in documentation

File: specs/http-api/README.md [hunk 5]
```diff
@@ -127,30 +125,13 @@ If the serviceEndpoint is itself a DID, this DID should resolve to a document an
...
+* Same Field; Multiple Values: `field=bar&field=baz`
```
Comment (angiejones) [inline specs/http-api/README.md:134]:
```suggestion
* Same Field; Multiple Values: `field=value&field=anotherValue`
```

File: specs/protocol/README.md [hunk 6]
```diff
@@ -838,9 +838,11 @@ sequenceDiagram
 	PFI1->>Alice: Offering
 	PFI2->>Alice: Offering
 	PFIn->>Alice: Offering
-  Alice->>PFI1: RequestForQuote (RFQ)
+  Alice->>PFI1: Request For Quote (RFQ)
```
Comment (angiejones) [inline specs/protocol/README.md:841]:
```suggestion
  Alice->>PFI1: RFQ (Request For Quote)
```

consistent with others here

File: specs/http-api/README.md (hunk 4 above)
Comment (jiyoontbd) [reply specs/http-api/README.md:133]: 😲 really! i did not know that. i quite like `foo bar baz` :D

File: specs/http-api/README.md (hunk 1 above)
Comment (decentralgabe) [reply specs/http-api/README.md:77]: something like a common base API https://swagger.io/docs/specification/api-host-and-base-path/

Comment (jiyoontbd) [reply specs/http-api/README.md]: done https://github.com/TBD54566975/tbdex/pull/370/commits/766aba6830c29d096fad65b79c39064c7af6cbb4

**Pull request merged**